table/meta_blocks.cc

```diff
@@ -223,7 +225,6 @@ bool NotifyCollectTableCollectorsOnFinish(
       for (const auto& prop : collector->GetReadableProperties()) {
         readable_properties.insert(prop);
       }
-      builder->Add(user_collected_properties);
```

Comment:
The table property block builder adds an invariant check that the same property is not re-added.

While `TablePropertiesCollector::Finish` API adds the user properties: https://github.com/facebook/rocksdb/blob/9058fd037c4ee6742ef60bec80f246f4dbd09ee9/include/rocksdb/table_properties.h#L129-L131

Adding the properties to the builder here will result in re-adding already added properties. So it is moved out of the for loop.

Reply:
Oh interesting ... did your new assertion fail in some UT? Is it because we can have same properties in `user_collected_properties` across collectors?

Reply:
Yeah, the new assertion failed in some external file ingestion test. And I found it's because this pattern here will do something like this:
Collector A inserts {"property_a": "value_a"} to the map => Adds the whole map, which include "property_a" to builder
Collector B inserts {"property_b": "value_b"} to the map => Adds the whole map which include "property_a" and "property_b" to the builder.